src/Opdex.Platform.Infrastructure/Clients/CirrusFullNodeApi/Modules/SmartContractsModule.cs

```diff
@@ -77,7 +78,21 @@ public async Task<LocalCallResponseDto> LocalCallAsync(LocalCallRequestDto reque
 
         using (_logger.BeginScope(logDetails))
         {
-            return await PostAsync<LocalCallResponseDto>(uri, httpRequest.Content, cancellationToken: cancellationToken);
+            LocalCallResponseDto response = null;
+            for (var retryCount = 0; retryCount < 3; retryCount++)
+            {
+                var callResult = await PostAsync<LocalCallResponseDto>(uri, httpRequest.Content, cancellationToken: cancellationToken);
+                if (callResult.HasError && callResult.ErrorMessage.Value == "No code at this address.")
+                {
+                    await Task.Delay(TimeSpan.FromMilliseconds(500 + (500 * retryCount)), cancellationToken);
+                    continue;
+                }
+
+                response = callResult;
+                break;
+            }
+
+            return response;
```

Comment:
Rather than setting `callResult` we might want to just set `response = await PostAsync...` If we fail 3 times in a row, we're going to return null which is going to cause null reference errors.

So 3 failures will still return the failed local call response with the error message

Reply:
Sorted